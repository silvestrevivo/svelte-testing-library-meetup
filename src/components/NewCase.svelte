<script>
  import CaseForm from "./CaseForm.svelte";

  function submitForm() {
    console.log("Form submitted");
  }
</script>

<div>
  <h1 data-testid="page-title">New Case</h1>
  <CaseForm {submitForm} />
</div>
