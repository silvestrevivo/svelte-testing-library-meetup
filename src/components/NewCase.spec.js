import { render, fireEvent } from '@testing-library/svelte';
import NewCase from './NewCase.svelte';

test('<NewCase />', () => {
  const { debug, getByTestId, queryByTestId, container } = render(NewCase);
  expect(getByTestId('page-title').textContent).toBe('New Case');
  //* we can render the component and the nested one
  debug();
  //* A big diference with react, is that first way is going to work
  expect(getByTestId('case-form')).toBeTruthy();
  expect(queryByTestId('case-form')).toBeTruthy();
  //* we don't need to shallow the component

  //* Snapshot test
  //* Container is the HTML information of the component => object
  //* This generates a folder with the snapshot/info of the component
  expect(container.firstChild).toMatchSnapshot();
})
