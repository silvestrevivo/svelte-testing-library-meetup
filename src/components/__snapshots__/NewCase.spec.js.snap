// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewCase /> 1`] = `
<div>
  <div>
    <h1
      data-testid="page-title"
    >
      New Case
    </h1>
     
    <form
      data-testid="case-form"
    >
      <input
        type="text"
      />
       
      <button>
        Submit
      </button>
    </form>
  </div>
</div>
`;
